app/services/foreman_bootdisk/iso_generator.rb

```diff
@@ -76,12 +88,35 @@ def self.generate(opts = {})
 
         if opts[:ipxe]
           ipxe_source_file = File.join(Setting[:bootdisk_ipxe_dir], 'ipxe.lkrn')
-          raise(Foreman::Exception, N_('Please ensure the ipxe-bootimgs package is installed.')) unless File.exist?(ipxe_source_file)
+          raise Foreman::Exception.new(N_('Please ensure the ipxe-bootimgs package is installed.')) unless File.exist?(ipxe_source_file)
 
           FileUtils.cp(ipxe_source_file, File.join(wd, 'build', 'ipxe'))
           File.open(File.join(wd, 'build', 'script'), 'w') { |file| file.write(opts[:ipxe]) }
         end
 
+        if opts[:grub]
+          grub_path = "/var/lib/tftpboot/grub2/grubx64.efi"
+          shim_path = "/var/lib/tftpboot/grub2/shimx64.efi"
+          raise Foreman::Exception.new(N_('Ensure %{path} exists by enabling TFTP smart proxy module.'), path: grub_path) unless File.exist?(grub_path)
```

Comment:
This exception was raised even though I had the tftp smart proxy module enabled. Either the message or puppet module/packaging needs to be fixed. Installing `foreman-bootloaders-redhat-tftpboot` made it go away.

Reply:
This RPM is only for downstream, running installer should do it: https://github.com/theforeman/puppet-foreman_proxy/blob/master/manifests/tftp/netboot.pp

Reply:
Anyway, I created new setting so this path is configurable. It will try to find grub in `/boot/efi/EFI` first and TFTP directory will be a fallback. This should now work out of box even when you don't have TFTP module enabled.